bogota-july-2018/Dominik Ortiz Galop/js/app.js

```diff
@@ -0,0 +1,139 @@
+/* ======= Model ======= */
+
+let model = {
+  number1: 0,
+  number2: 0
+};
+
+/* ======= Controller ======= */
+
+const controller = {
+  setNumber(modelNumber, value) {
+    model[modelNumber] = value;
+  },
+
+  reset() {
+    model = {
+      number1: 0,
+      number2: 0
+    };
+  },
+
+  getNumber(number) {
+    return model[number];
+  },
+
+  sum(numb1, numb2) {
+    return numb1 + numb2;
+  },
+
+  rest(numb1, numb2) {
+    return numb1 - numb2;
+  },
+
+  multiplication(numb1, numb2) {
+    return numb1 * numb2;
+  },
+
+  division(numb1, numb2) {
+    return parseFloat(numb1 / numb2);
+  },
+
+  percentage(numb1, per) {
+    return parseFloat((numb1 / 100) * per);
+  },
+
+  changeSymbol(modelNumber, symbol) {
+    const change = symbol
+      ? Math.abs(model[modelNumber])
+      : -Math.abs(model[modelNumber]);
+    model[modelNumber] = change;
+    return change;
+  }
+};
+
+/* ======= View ======= */
```

Comment:
Not sure if there's some patter with a view in JS, (i would say NO), this is actually how a controller looks, or a VIEW MODEL in MVVM pattern.

Reply:
For this design pattern i use has a reference this 3 guide, curse or book to JavaScript Design Patterns

https://addyosmani.com/resources/essentialjsdesignpatterns/book/#detailmvc

https://www.toptal.com/javascript/comprehensive-guide-javascript-design-patterns

https://www.udacity.com/course/javascript-design-patterns--ud989 (i completed this curse in the past and they teach me about mvc, in this video they introduce me to MVChttps://www.youtube.com/watch?time_continue=124&v=AhyrzeU9LGI&ab_channel=Udacity  )